### ISSUE TITLE
ReduceLRonPlateau does not work as expected

### ISSUE THREAD

**Issue by vykhand** [skorch-dev/skorch#363]
Hi everyone!

I noticed that the LR Scheduler callback with ReduceLRonPlateau does not work as expected. I expect it to reduce LR when the metric has not improved for N=**patience** epochs, but instead it reduces LR just after the fixed N number of epochs

For example:
(config.yaml)
```
scheduler_params: {factor: 0.5, min_lr: 1.0e-06, mode: max, patience: 15, verbose: 2}
```
(NeuralNet callback)
```
scheduler = LRScheduler(policy=scheduler_policy, monitor="val_ioi_best", **model_params["scheduler_params"])
```
or 
```
scheduler = LRScheduler(policy=scheduler_policy, monitor="val_iou", **model_params["scheduler_params"])
```

```
TRAIN PHASE ITER: 3
, global time: 3.29, mem use:0.44GB, GPU_MEM: 0.00GB
  epoch    train_iou    train_loss    val_iou    valid_loss    cp       dur
```
-------  -----------  ------------  ---------  ------------  ----  --------
      1       0.4349        0.2884     0.5459        0.2219     +  223.1961
      2       0.6177        0.1862     0.5850        0.2172     +  222.8284
      3       0.6567        0.1597     0.6741        0.1751     +  222.5643
      4       0.6951        0.1388     0.7109        0.1692     +  222.2332
      5       0.7023        0.1326     0.6809        0.2199        222.3694
      6       0.7277        0.1271     0.6794        0.1622     +  222.0340
      7       0.7418        0.1085     0.6894        0.1746        221.9729
      8       0.7542        0.1038     0.7041        0.1831        221.9098
      9       0.7776        0.0912     0.7253        0.1932        221.9841
     10       0.7814        0.0860     0.7434        0.1550     +  221.7720
     11       0.7841        0.0830     0.7262        0.1764        221.8531
     12       0.7972        0.0771     0.7581        0.1676        222.0086
     13       0.8117        0.0679     0.7438        0.1924        221.5925
     14       0.8099        0.0727     0.7128        0.2350        221.8342
     15       0.8027        0.0736     0.7150        0.2062        222.1696
     16       0.8042        0.0729     0.7447        0.1813        221.6657
Epoch    16: reducing learning rate of group 0 to 5.0000e-05.

**Comment by Dewald928** [comment]
I don't know if this is a user error, but this isn't working as intended. It keeps changing the LR after patience has run out every 10 epochs.
When I debugged the code, it seems like the "score" = -np.inf indefinitely. 

I made a workaround, although I don't think it's perfect.

```
if isinstance(self.lr_scheduler_, ReduceLROnPlateau):
            if callable(self.monitor):
                score = self.monitor(net)
            else:
                if len(net.history) > 0:
                    score = net.history[-1, self.monitor]
                elif self.lr_scheduler_.mode == 'max':
                    score = -np.inf
                elif self.lr_scheduler_.mode == 'min':
                    score = np.inf

            self.lr_scheduler_.step(score)
```

**Comment by ottonemo** [comment]
@Dewald928 can you give a bit more context to this? How do you initialize the callback? Do you have an example that reproduces the behavior?

**Comment by Dewald928** [comment]
Initially I set up my callback as
```
lrscheduler = LRScheduler(policy=lr_scheduler.ReduceLROnPlateau,
                                          monitor='valid_tss',
                                          mode='max',
                                          factor=0.5,
                                          verbose=True)
```
But having the mode='max' causes the 'score' variable to be -np.inf for every epoch, which causes the patience to run out, without ever plateauing. 

The first epoch should be -np.inf. But afterwards it should be the monitored value.

Am I using it wrong? Because my crude fix makes it work as expected.

**Comment by BenjaminBossan** [comment]
The proposed change by @Dewald928 looks much more correct to me than the current implementation. If not to say, the current code looks very wrong to me.

We might even change the code [here](https://github.com/skorch-dev/skorch/blob/be5569d31755bd375ebf44e47d66539e815fff13/skorch/callbacks/lr_scheduler.py#L151-L157) to just:

```python
            else:
                score = net.history[-1, self.monitor]
```

It seems that the +/- inf is only required if the order of the callbacks is incorrect, i.e. if the lr scheduler is placed _before_ the callback that creates the score (arguably we should error out there to avoid an off by 1 error).

**Unrelated**: The `simulate` method does not work with `ReduceLROnPlateau` as that would require a metric to monitor. How about just passing `1.0` (or whatever constant) in that case, or have the user pass the metric as a parameter?
